packages/services/src/view-service/delegations-by-address-index.test.ts

```diff
@@ -300,35 +305,121 @@ describe('DelegationsByAddressIndex request handler', () => {
+      it('returns one `ValueView` for each active validator', async () => {
+        const results: (
+          | DelegationsByAddressIndexResponse
+          | PartialMessage<DelegationsByAddressIndexResponse>
+        )[] = [];
+
+        for await (const result of delegationsByAddressIndex(
+          new DelegationsByAddressIndexRequest({ addressIndex: { account: 0 } }),
+          mockCtx,
+        )) {
+          results.push(result);
+        }
+
+        expect(results.length).toBe(2);
+      });
+
+      it('returns a zero-balance `ValueView` for validators the address has no tokens for', async () => {
+        const results: (
+          | DelegationsByAddressIndexResponse
+          | PartialMessage<DelegationsByAddressIndexResponse>
+        )[] = [];
+
+        for await (const result of delegationsByAddressIndex(
+          new DelegationsByAddressIndexRequest({ addressIndex: { account: 0 } }),
+          mockCtx,
+        )) {
+          results.push(result);
+        }
+
+        const secondValueView = new ValueView(results[1]!.valueView);
+
+        expect(getAmount(secondValueView)).toEqual({ hi: 0n, lo: 0n });
+      });
+    });
 
-      for await (const result of delegationsByAddressIndex(
-        new DelegationsByAddressIndexRequest({
-          addressIndex: { account: 0 },
-          filter: DelegationsByAddressIndexRequest_Filter.ALL_ACTIVE_WITH_NONZERO_BALANCES,
-        }),
-        mockCtx,
-      )) {
-        results.push(result);
-      }
+    describe('when the nonzero balances filter option is passed', () => {
+      it('returns one `ValueView` for each validator the address has tokens for', async () => {
+        const results: (
+          | DelegationsByAddressIndexResponse
+          | PartialMessage<DelegationsByAddressIndexResponse>
+        )[] = [];
+
+        for await (const result of delegationsByAddressIndex(
+          new DelegationsByAddressIndexRequest({
+            addressIndex: { account: 0 },
+            filter: DelegationsByAddressIndexRequest_Filter.ALL_ACTIVE_WITH_NONZERO_BALANCES,
+          }),
+          mockCtx,
+        )) {
+          results.push(result);
+        }
+
+        expect(results.length).toBe(1);
+      });
+    });
 
-      expect(results.length).toBe(1);
+    describe('when the `ALL` filter option is passed', () => {
+      it('returns one `ValueView` for each validator, including inactive ones', async () => {
+        const results: (
+          | DelegationsByAddressIndexResponse
+          | PartialMessage<DelegationsByAddressIndexResponse>
+        )[] = [];
+
+        for await (const result of delegationsByAddressIndex(
+          new DelegationsByAddressIndexRequest({
+            addressIndex: { account: 0 },
+            filter: DelegationsByAddressIndexRequest_Filter.ALL,
+          }),
+          mockCtx,
+        )) {
+          results.push(result);
+        }
+
+        expect(results.length).toBe(4);
+      });
     });
   });
```

Comment:
No changes to the tests above, just nested them in a describe block so the mocked functions do not conflict with the new test